Merged the content with Benefits of MSAL

Rinki-MSFT opened the pull request:
Based on an email discussion, merging [Benefits of MSAL.docx](https://microsoft.sharepoint-df.com/:w:/t/IdentityCxPDeveloperCare/EbouexXd2W5Jn8wRKagY4BwBAC91HDeTjPjUZXWQcAokVg?e=copkhy) with the concept page. Details here - [Task 2161746](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2161746): [Documentation][MSAL]Landing Page explaining 'Why MSAL' is needed to start S500 Customer Engagement

Rinki-MSFT committed:
Merged the content with Benefits of MSAL

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -61,6 +77,15 @@ After identifying your apps that use ADAL, migrate them to MSAL depending on you
+MSAL Supports a wide range of application types and scenarios. Please refer to [Microsoft Authentication Library support for several application types](https://learn.microsoft.com/en-us/azure/active-directory/develop/reference-v2-libraries#single-page-application-spa).
+
+ADAL to MSAL Migration Guide for different platforms are available in the following link.
+- [Migrate to MSAL iOS and MacOS](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-objc-adal-msal)
```
```suggestion
- [Migrate to MSAL iOS and MacOS](migrate-objc-adal-msal.md)
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -61,6 +77,15 @@ After identifying your apps that use ADAL, migrate them to MSAL depending on you
+
+ADAL to MSAL Migration Guide for different platforms are available in the following link.
+- [Migrate to MSAL iOS and MacOS](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-objc-adal-msal)
+- [Migrate to MSAL Java](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-adal-msal-java)
```
```suggestion
- [Migrate to MSAL Java](migrate-adal-msal-java.md)
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -61,6 +77,15 @@ After identifying your apps that use ADAL, migrate them to MSAL depending on you
+ADAL to MSAL Migration Guide for different platforms are available in the following link.
+- [Migrate to MSAL iOS and MacOS](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-objc-adal-msal)
+- [Migrate to MSAL Java](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-adal-msal-java)
+- [Migrate to MSAL .Net](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-net-migration)
```
```suggestion
- [Migrate to MSAL .Net](msal-net-migration.md)
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -61,6 +77,15 @@ After identifying your apps that use ADAL, migrate them to MSAL depending on you
+- [Migrate to MSAL iOS and MacOS](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-objc-adal-msal)
+- [Migrate to MSAL Java](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-adal-msal-java)
+- [Migrate to MSAL .Net](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-net-migration)
+- [Migrate to MSAL Node](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-node-migration)
```
```suggestion
- [Migrate to MSAL Node](msal-node-migration.md)
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -61,6 +77,15 @@ After identifying your apps that use ADAL, migrate them to MSAL depending on you
+- [Migrate to MSAL Java](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-adal-msal-java)
+- [Migrate to MSAL .Net](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-net-migration)
+- [Migrate to MSAL Node](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-node-migration)
+- [Migrate to MSAL Python](https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal)   
```
```suggestion
- [Migrate to MSAL Python](migrate-python-adal-msal.md)   
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -61,6 +77,15 @@ After identifying your apps that use ADAL, migrate them to MSAL depending on you
 
 [!INCLUDE [application type](includes/adal-msal-migration.md)]
 
+MSAL Supports a wide range of application types and scenarios. Please refer to [Microsoft Authentication Library support for several application types](https://learn.microsoft.com/en-us/azure/active-directory/develop/reference-v2-libraries#single-page-application-spa).
```
```suggestion
MSAL Supports a wide range of application types and scenarios. Please refer to [Microsoft Authentication Library support for several application types](reference-v2-libraries.md#single-page-application-spa).
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -46,6 +55,13 @@ MSAL provides multiple benefits over ADAL, including the following features:
 | Throttling |![Throttling - MSAL provides the feature][y]|![Throttling - ADAL doesn't provide the feature][n]|
 
+## Additional Capabilities of MSAL over ADAL
+- Auth Broker support – Device based Conditional Access Policy. 
```
```suggestion
- Auth broker support – Device-based Conditional Access policy
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -46,6 +55,13 @@ MSAL provides multiple benefits over ADAL, including the following features:
 
+## Additional Capabilities of MSAL over ADAL
+- Auth Broker support – Device based Conditional Access Policy. 
+- Proof of Possession tokens.
```
```suggestion
- Proof of possession tokens
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -46,6 +55,13 @@ MSAL provides multiple benefits over ADAL, including the following features:
+## Additional Capabilities of MSAL over ADAL
+- Auth Broker support – Device based Conditional Access Policy. 
+- Proof of Possession tokens.
+- Azure AD Certificate-based Authentication (CBA) on Mobile.
```
```suggestion
- Azure AD certificate-based authentication (CBA) on mobile
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -46,6 +55,13 @@ MSAL provides multiple benefits over ADAL, including the following features:
+- Auth Broker support – Device based Conditional Access Policy. 
+- Proof of Possession tokens.
+- Azure AD Certificate-based Authentication (CBA) on Mobile.
+- System browsers on mobile devices.
```
```suggestion
- System browsers on mobile devices
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -46,6 +55,13 @@ MSAL provides multiple benefits over ADAL, including the following features:
+- Proof of Possession tokens.
+- Azure AD Certificate-based Authentication (CBA) on Mobile.
+- System browsers on mobile devices.
+- Where ADAL had only Authentication Context Class, MSAL exposes the notion of has a collection of client apps (public client and Confidential Client).
```
```suggestion
- Where ADAL had only authentication context class, MSAL exposes the notion of a collection of client apps (public client and confidential client).
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
 
 ## Why switch to MSAL?
 
+To understand 'Why MSAL?', it's important to first understand the differences between Microsoft identity platform (v2.0) and Azure Active Directory (v1.0) endpoints. The v1.0 endpoint is used by ADAL while the v2.0 endpoint is used by MSAL. If you have developed apps against the v1 endpoint in the past, you would probably be using ADAL (Azure AD authentication Library). Since the v2 endpoint has  ...
```
```suggestion
To understand 'Why MSAL?', it's important to first understand the differences between Microsoft identity platform (v2.0) and Azure Active Directory (v1.0) endpoints. The v1.0 endpoint is used by Azure AD Authentication Library (ADAL) while the v2.0 endpoint is used by Microsoft Authentication Library (MSAL). If you've developed apps against the v1.0 endpoint in the past, you're likely using ADAL. Since the v2.0 endpoint has changed significantly enough, the new library (MSAL) was built for the new endpoint entirely. 
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
 
+To understand 'Why MSAL?', it's important to first understand the differences between Microsoft identity platform (v2.0) and Azure Active Directory (v1.0) endpoints. The v1.0 endpoint is used by ADAL while the v2.0 endpoint is used by MSAL. If you have developed apps against the v1 endpoint in the past, you would probably be using ADAL (Azure AD authentication Library). Since the v2 endpoint has  ...
+
+The following diagram shows the v2 vs v1 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
```
```suggestion
The following diagram shows the v2.0 vs v1.0 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
+The following diagram shows the v2 vs v1 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
+![image](https://user-images.githubusercontent.com/76989215/202167545-5777c943-b68b-46cb-95ed-3416cbf5f2c1.png)
+
+MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](https://user-images.githubusercontent.com/76989215/202164286-9b07d541-b48b-4297-9dd6-69927e1edc69.png)
```
These are not the original assets. Looking for them now.

CelesteDG reviewed (changes requested):
Hi @Rinki-MSFT - Please incorporate the changes I made to the PR. Thank you.

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
+To understand 'Why MSAL?', it's important to first understand the differences between Microsoft identity platform (v2.0) and Azure Active Directory (v1.0) endpoints. The v1.0 endpoint is used by ADAL while the v2.0 endpoint is used by MSAL. If you have developed apps against the v1 endpoint in the past, you would probably be using ADAL (Azure AD authentication Library). Since the v2 endpoint has  ...
+
+The following diagram shows the v2 vs v1 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
+![image](https://user-images.githubusercontent.com/76989215/202167545-5777c943-b68b-46cb-95ed-3416cbf5f2c1.png)
```
```suggestion
![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
```

CelesteDG commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
+
+MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](https://user-images.githubusercontent.com/76989215/202164286-9b07d541-b48b-4297-9dd6-69927e1edc69.png)
+
+MSAL is designed to enable a secure solution without developers having to worry about the implementation details. it simplifies and manages acquiring, managing, caching, and refreshing tokens, and uses best practices for resilience. It is recommended to use MSAL to [increase the resilience of authentication and authorization in client applications you develop](https://learn.microsoft.com/en-us/az ...
```
```suggestion
MSAL is designed to enable a secure solution without developers having to worry about the implementation details. it simplifies and manages acquiring, managing, caching, and refreshing tokens, and uses best practices for resilience. We recommend you use MSAL to [increase the resilience of authentication and authorization in client applications that you develop](../fundamentals/resilience-client-app.md?tabs=csharp#use-the-microsoft-authentication-library-msal).
```

CelesteDG reviewed (changes requested):
@Rinki-MSFT Some additional asks to update/accept my changes. Thank you.

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

Rinki-MSFT committed:
Update articles/active-directory/develop/msal-migration.md

Co-authored-by: Celeste de Guzman <16906646+CelesteDG@users.noreply.github.com>

jmprieur reviewed (approved):
Lgtm. Thanks @Rinki-MSFT and @CelesteDG

Jak-MS commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
+To understand 'Why MSAL?', it's important to first understand the differences between Microsoft identity platform (v2.0) and Azure Active Directory (v1.0) endpoints. The v1.0 endpoint is used by Azure AD Authentication Library (ADAL) while the v2.0 endpoint is used by Microsoft Authentication Library (MSAL). If you've developed apps against the v1.0 endpoint in the past, you're likely using ADAL. ...
+
+The following diagram shows the v2.0 vs v1.0 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
+![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
```
```suggestion
The following diagram shows the v2.0 vs v1.0 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.

![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
```

added a line break to fix rendering... also alt text needs to be fixed for the diagram here.

Jak-MS commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,15 @@ If any of your applications use the Azure Active Directory Authentication Librar
+The following diagram shows the v2.0 vs v1.0 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
+![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
+
+MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](https://user-images.githubusercontent.com/76989215/202164286-9b07d541-b48b-4297-9dd6-69927e1edc69.png)
```
```suggestion
MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](https://user-images.githubusercontent.com/76989215/202164286-9b07d541-b48b-4297-9dd6-69927e1edc69.png)
```

the link in this line opens an image that is just the text... this needs to be fixed.

Rinki-MSFT committed:
Update msal-migration.md

Corrected link in line#38. Added a new line after line#34 before the image as the alignment was getting disturbed otherwise.

Court72 commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,16 @@ If any of your applications use the Azure Active Directory Authentication Librar
+
+![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
+
+MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](https://learn.microsoft.com/en-us/azure/active-directory/azuread-dev/azure-ad-endpoint-comparison)
```
```suggestion
MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](../azuread-dev/azure-ad-endpoint-comparison.md)
```

Court72 committed:
fix link warning

Court72 commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,16 @@ If any of your applications use the Azure Active Directory Authentication Librar
+
+The following diagram shows the v2.0 vs v1.0 endpoint experience at a high level, including the app registration experience, SDKs, endpoints, and supported identities.
+
+![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
```
```suggestion
![Diagram that shows the v1.0 versus the v2.0 architecture.](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
```

Court72 commented on articles/active-directory/develop/msal-migration.md:
```diff
@@ -29,6 +29,16 @@ If any of your applications use the Azure Active Directory Authentication Librar
+
+![v1.0 vs v2.0 architecture](../azuread-dev/media/about-microsoft-identity-platform/about-microsoft-identity-platform.svg)
+
+MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](../azuread-dev/azure-ad-endpoint-comparison.md)
```
```suggestion
MSAL leverages all the [benefits of Microsoft identity platform (v2.0) endpoint](../azuread-dev/azure-ad-endpoint-comparison.md).
```

Court72 committed:
Apply suggestions from PR review
